src/main/java/bio/terra/service/snapshotbuilder/utils/AggregateSynapseQueryResultsUtils.java

```diff
@@ -38,4 +38,14 @@ public static int toCount(ResultSet rs) {
           "Error processing result set into SnapshotBuilderConcept model", e);
     }
   }
+
+  public static String toDomainId(ResultSet rs) {
+    try {
+      // Azure ResultSet is 1 indexed
+      // https://docs.oracle.com/javase/7/docs/api/java/sql/ResultSet.html
+      return rs.getString(1);
+    } catch (SQLException e) {
+      throw new ProcessResultSetException("Error processing result set into String domain ID", e);
```

Comment:
It looks like all methods need to do this. Could it be done in the caller instead?

Alternatively, you could refactor the common code.  Something like
```

  static <T> T getFirst(CountGetter<T> getter) {
    try {
      // Java ResultSet is 1 indexed
      // https://docs.oracle.com/en/java/javase/17/docs/api/java.sql/java/sql/ResultSet.html#getInt(int)
      return getter.get(1);
    } catch (SQLException e) {
      throw new ProcessResultSetException("Error processing result set", e);
    }
  }

  public static int toCount(ResultSet rs) {
    return getFirst(rs::getInt);
  }

  public static String toDomainId(ResultSet rs) {
    return getFirst(rs::getString);
  }
```

Reply:
I agree it would be nice to have this repeated less. If we try to propagate the error up a level, it would be caught in the AzureSynapsePdao.runQuery method under the DataAccessException and we would lose some specificity of the error message. Maybe I could add a check for a SQLException specifically there? Or maybe there is a way to propagate it further up to runQuery? But I'm not sure.